datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -219,6 +219,9 @@ public class Coordinator implements ZkAdapter.ZkAdapterListener, MetricsAware {
   private final Map<String, SerdeAdmin> _serdeAdmins = new HashMap<>();
   private final Map<String, Authorizer> _authorizers = new HashMap<>();
   private volatile boolean _shutdown = false;
+  // TODO we have _shutdown, eventThread and now _handleEventCompleted, for some distinct usage,
+  //  we should revisit and refactor to have less variation
+  private volatile boolean _handleEventCompleted = false;
```

Comment:
nit: instead of `_handleEventCompleted ` can we name this variable to something like, `_coordinatorEventThreadExiting` ?

Reply:
thought about this back and forth, ideally we just need to wait for the handleEvent to get the lock on the coordinator object, once we are out of it, zk expiry or main thread can get another chance to acquire lock. coordintorEventThread exists only after we get out of the run method, which may not be the case.

nothing will be exactly certain here, so tried to came up with the one which is more closer to the logic.